database/postgres/delivery_attempts.go

```diff
@@ -0,0 +1,135 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"github.com/frain-dev/convoy/database"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/jmoiron/sqlx"
+	"io"
+	"time"
+)
+
+type deliveryAttemptRepo struct {
+	db *sqlx.DB
+}
+
+func NewDeliveryAttemptRepo(db database.Database) datastore.DeliveryAttemptsRepository {
+	return &deliveryAttemptRepo{
+		db: db.GetDB(),
+	}
+}
+
+var (
+	_ datastore.DeliveryAttemptsRepository = (*deliveryAttemptRepo)(nil)
+)
+
+const (
+	creatDeliveryAttempt = `
+    INSERT INTO convoy.delivery_attempts (id, url, method, api_version, endpoint_id, event_delivery_id, project_id, ip_address, request_http_header, response_http_header, http_status, response_data, error, status)
+    VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14);
+    `
+
+	softDeleteProjectDeliveryAttempts = `
+    UPDATE convoy.delivery_attempts SET deleted_at = NOW() WHERE project_id = $1 AND created_at >= $2 AND created_at <= $3 AND deleted_at IS NULL;
+    `
+
+	hardDeleteProjectDeliveryAttempts = `
+    DELETE FROM convoy.delivery_attempts WHERE project_id = $1 AND created_at >= $2 AND created_at <= $3 AND deleted_at IS NULL;
+    `
+
+	findDeliveryAttempts = `with att as (SELECT * FROM convoy.delivery_attempts WHERE event_delivery_id = $1 order by created_at desc limit 10) select * from att order by created_at;`
```

Comment:
Hm, I'm confused why the CTE is needed here

Reply:
We need to fetch the newest items but to be backward compatible; they need to be ordered in ASC order.

Reply:
Okay cool!